crates/services/gas_price_service/src/v1/service.rs

```diff
@@ -32,94 +44,85 @@ use fuel_core_services::{
     StateWatcher,
     TaskNextAction,
 };
+use fuel_core_types::fuel_types::BlockHeight;
 use fuel_gas_price_algorithm::{
     v0::AlgorithmUpdaterV0,
     v1::{
         AlgorithmUpdaterV1,
         AlgorithmV1,
+        UnrecordedBlocks,
     },
 };
 use futures::FutureExt;
-use std::num::NonZeroU64;
-use tokio::sync::broadcast::{
-    error::RecvError,
-    Receiver,
-};
-
-use crate::common::utils::Result as GasPriceResult;
+use tokio::sync::broadcast::Receiver;
 
 /// The service that updates the gas price algorithm.
-pub struct GasPriceServiceV1<L2, Metadata, DA>
+pub struct GasPriceServiceV1<L2, DA, StorageTxProvider>
 where
     DA: DaBlockCostsSource,
+    StorageTxProvider: TransactionableStorage,
 {
     /// The algorithm that can be used in the next block
     shared_algo: SharedV1Algorithm,
     /// The L2 block source
     l2_block_source: L2,
-    /// The metadata storage
-    metadata_storage: Metadata,
     /// The algorithm updater
     algorithm_updater: AlgorithmUpdaterV1,
     /// the da source adapter handle
     da_source_adapter_handle: DaSourceService<DA>,
     /// The da source channel
     da_source_channel: Receiver<DaBlockCosts>,
+    /// Buffer of block costs from the DA chain
+    da_block_costs_buffer: Vec<DaBlockCosts>,
+    /// Storage transaction provider for metadata and unrecorded blocks
+    pub(crate) storage_tx_provider: StorageTxProvider,
```

Comment:
Seems like this is `pub(crate)` only because of tests. Maybe we can add a test-only getter?